### ISSUE TITLE
Changing the license

### ISSUE THREAD

**Issue by tomasklaen** [tomasklaen/uosc#682]
To my understanding the current license (GPL) forces any project that uses uosc to inherit it as well, making it incredibly restrictive.

LGPL on the other hand means that anyone using uosc doesn't have to inherit its license, they just have to credit it, and provide source if they do any modifications to it specifically. This is way more reasonable, so I'd want to change the license to LGPL to allow people to use uosc in more places.

For that, I guess I need approval of every contributor since this project has been licensed under GPL. So if you guys could confirm that you **approve of this change ~~and transfer the copyright to the code you contributed to me (Tomas Klaen)~~**, that would be great:

- [x] @christoph-heinrich 
- [x] @po5
- [x] @dyphire
- [x] @natural-harmonia-gropius
- [x] @hooke007
- [x] @Hill-98
- [x] @microraptor
- [x] @shvchk
- [ ] @evantj
- [ ] @unavailable500
- [x] @sarmong
- [x] @larvanitis

I guess just post:

> Yes, I approve.

or something along those lines. ~~The transfer of copyright is necessary so that I don't have to do this again in the future if it turns out LGPL is messing something up as well, and can re-license without bothering you again. I'm not a lawyer and I doubt the LGPL license summaries I've found covered all of the possible implications.~~

To be a bit more reasonable about this, I've left out people whose contribution(s) no longer seem to be part of uosc (overwritten or removed), or people who contributed only stuff like typo fixes, documentation tweaks, or configuration adjustments.

In case someone won't reply or approve, I'll either try to remove/replace their contributions, or stay with GPL, depending on how severe of an issue that would be.

**Comment by natural-harmonia-gropius** [comment]
Yes, I approve.

**Comment by hooke007** [comment]
Yes, I approve.

**Comment by dyphire** [comment]
Yes, I approve.

**Comment by microraptor** [comment]
Yes, I approve.

**Comment by sarmong** [comment]
Yes, I approve.

**Comment by christoph-heinrich** [comment]
I assume you mean the LGPL 2.1? As that is the only one Github provides.

Whichever it is, I approve the re-license.

HF bothering me about future license changes :P

**Comment by Hill-98** [comment]
Yes, I approve.

**Comment by shvchk** [comment]
Yes, I approve.

**Comment by po5** [comment]
It seems like the giving up of copyright would only work if you also added a clause covering future contributions.

I approve the re-license to whatever license you choose now or in the future, it can't do any harm anyway since older versions will still be available under GPL.

**Comment by larvanitis** [comment]
Yes, I approve.

**Comment by tomasklaen** [comment]
> It seems like the giving up of copyright would only work if you also added a clause covering future contributions.

Yeah that's the plan, just need to figure out how to do it. I guess a disclaimer in a PR template or something.

> I assume you mean the LGPL 2.1? As that is the only one Github provides.

Well now I have to check what the difference is between 2.1 and 3, and why github doesn't provide 3 :(

> HF bothering me about future license changes :P

So that's a yes to re-license, but no to copyright transfer? If you're afraid of me doing something crazy, you can always just fork the previous commit and unstar the repo :) But fair enough either way. I just didn't want to go through this again if the need arises, but 1 active contributor to ask is not a big deal.

**Comment by tomasklaen** [comment]
After another hour of reading seemingly contradictory assessments of LGPL restrictions and comparisons between 2.1 and 3, I think I'll just inherit LGPLv2.1 of mpv and be done with it. It makes sense for it to have the same license anyway. No copyright transfer requests or anything.

**Comment by tomasklaen** [comment]
Coincidentally the two people who didn't respond yet both contributed only scripts or tweaks to install uosc from command line, which is exactly what I'm rewriting from scratch now, so I guess we're good to go here.

**Comment by christoph-heinrich** [comment]
> If you're afraid of me doing something crazy

It's already bad enough that companies get complete ownership over code I write, I want to at least own the stuff I write in my spare time. Otherwise there can be the bizarre situation where I want to reuse some code I wrote in a project that isn't license compatible, but since I don't own it anymore I have to actually ask permission to use the code I wrote myself. I don't doubt that you would actually give me permission, should that situation ever arise, but I don't like the principle of it.
